src/mongodb-migrations.ts

```diff
@@ -73,9 +62,7 @@ export class Migrator {
     this._m = [];
     this._result = {};
 
-    this._dbReady = (Promise.fromCallback as any)((cb: (error?: Error) => void) => {
-      mongoConnect(dbConfig, cb);
-    }).then((client: MongoClient) => {
+    this._dbReady = mongoConnect(dbConfig).then((client: MongoClient) => {
```

Comment:
This is an interesting pattern I preserved from the original. The constructor can't be async, but it starts some async mongo client initialization. `_dbReady` is a promise that resolves when initialization is done, and then every other method is `async` and awaits on it first.